inc/functions.php

```diff
@@ -1906,28 +1906,295 @@ function my_set_array_cookie($name, $id, $value, $expires="")
 	$mybb->cookies['mybb'][$name] = $newcookie;
 }
 
+/*
+ * Arbitrary limits for _safe_unserialize()
+ */
+define('MAX_SERIALIZED_INPUT_LENGTH', 4096);
+define('MAX_SERIALIZED_ARRAY_LENGTH', 256);
+define('MAX_SERIALIZED_ARRAY_DEPTH', 5);
```

Comment:
@Destroy666x I left them there because on boards with small stack this may be a problem so reducing the array length and depth allowed will stop the stack from filling up to a max.